ydb/core/security/ticket_parser_impl.h

```diff
@@ -644,6 +656,36 @@ class TTicketParserImpl : public TActorBootstrapped<TDerived> {
         return false;
     }
 
+    template <typename TTokenRecord>
+    bool CanInitTokenFromCertificate(const TString& key, TTokenRecord& record) {
+        if (record.TokenType != TDerived::ETokenType::Certificate) {
+            return false;
+        }
+        const static TString error = "Cannot create token from certificate. Cannot extract subject from certificate";
+        CounterTicketsCertificate->Inc();
+        X509CertificateReader::X509Ptr x509cert = X509CertificateReader::ReadCertAsPEM(record.Certificate);
+        if (!x509cert) {
+            SetError(key, record, { .Message = error, .Retryable = false });
+            return false;
+        }
+        TStringBuilder dn;
+        for (const auto& [attribute, value] : X509CertificateReader::ReadAllSubjectTerms(x509cert)) {
+            dn << attribute << "=" << value << ",";
+        }
+        if (dn.empty()) {
+            SetError(key, record, { .Message = error, .Retryable = false });
```

Comment:
it's better to have different error messages in different cases

Reply:
done